### PR TITLE
3 get api reviews

### PR DESCRIPTION
**Pull request by ahsan-0**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -19,3 +19,37 @@ describe("GET /api/categories", () => {
...
+      .get("/api/reviews")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.reviews[5]).toEqual({
+          review_id: 2,
+          title: "Jenga",
+          category: "dexterity",
+          designer: "Leslie Scott",
+          owner: "philippaclaire9",
+          review_body: "Fiddly fun for all the family",
```
Comment (popatre) [inline __tests__/app.test.js]: The ticket doesn't want the body on there.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -19,3 +19,37 @@ describe("GET /api/categories", () => {
...
   });
 });
+
+describe("GET /api/reviews", () => {
+  test("should respond with 200 and get all reviews with an added key of comment count", () => {
+    return request(app)
+      .get("/api/reviews")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.reviews[5]).toEqual({
```
Comment (popatre) [inline __tests__/app.test.js]: could you refactor this to use a forEach - then check each object matches the shape you want, using .toMatchObject - or something similar?
This is a brittle test at the moment, is the reviews were to change

File: __tests__/app.test.js [hunk 3]
```diff
@@ -19,3 +19,37 @@ describe("GET /api/categories", () => {
...
+          review_id: 2,
+          title: "Jenga",
```
Comment (popatre) [inline __tests__/app.test.js]: Could you also check the length of the array, to make sure you're getting back all the reviews in the dataset?

File: models/models.js [hunk 4]
```diff
@@ -4,3 +4,13 @@ exports.selectCategories = () => {
     return rows;
   });
 };
+
+exports.selectReviews = () => {
+  return db
+    .query(
+      `SELECT reviews.*, COUNT(comments.review_id) AS comment_count FROM reviews LEFT JOIN comments ON reviews.review_id = comments.review_id GROUP BY reviews.review_id;`
```
Comment (popatre) [inline models/models.js]: Could you adjust this to not include the body?

Comment (popatre) [reply models/models.js]: Nice join apart from that

File: models/models.js [hunk 5]
```diff
@@ -4,3 +4,13 @@ exports.selectCategories = () => {
...
+    )
+    .then(({ rows }) => {
```
Comment (popatre) [inline models/models.js:13]: Good destructuring.

**Review by popatre — CHANGES_REQUESTED** [review]
Could you refactor the test then send it in again for someone to look at?

File: __tests__/app.test.js [hunk 6]
```diff
@@ -15,7 +15,55 @@ describe("GET /api/categories", () => {
...
+        });
+      });
+  });
+  test("review array should have a length of 13", () => {
+    return request(app)
+      .get("/api/reviews")
+      .expect(200)
+      .then(({ body }) => {
+        const { reviews } = body;
+        expect(reviews).toHaveLength(13);
```
Comment (Poonam-raj) [inline __tests__/app.test.js:67]: Hi Ahsan, we also need this list of reviews sorted by date in descending order - we need a assertion to check this is the case.

File: __tests__/app.test.js [hunk 7]
```diff
@@ -15,7 +15,55 @@ describe("GET /api/categories", () => {
...
+            review_id: expect.any(Number),
+            title: expect.any(String),
+            owner: expect.any(String),
+            category: expect.any(String),
+            review_img_url: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            designer: expect.any(String),
+            comment_count: expect.any(String),
+          });
```
Comment (Poonam-raj) [inline __tests__/app.test.js:56]: A nice additional assertion to backup the need for 'review_body' to not be here is to write an assertion specifically saying that this property should not be present in each review

**Review by Poonam-raj — APPROVED** [review]
Hi Ahsan, again nice work on this. A few small amendments needed in your testing for this to be a watertight GET reviews endpoint. 

I'd really like you to implement the sorting assertion before this PR can go through as it is on the ticket. Once you've done that and it's passing I'm fine with you merging this work

**Pull request merged**